### PR TITLE
Move enarx_keep_sgx::{Builder, Enclave} into sgx crate

### LINKED ISSUE #754: Make sgx::ioctls module private
## Description

As reported by @npmccallum in #746:

Expose a high-level interface. We already have an imperfect one in `enarx-keep-sgx::builder`. We could move that from `enarx-keep-sgx` to `sgx`. Then `sgx::ioctls` could be made private.

### PR DESCRIPTION
**Pull request by kubkon**
This commit extracts `enarx_keep_sgx::{Builder, Enclave}`  and other public-facing structs into the `sgx` crate. The structs
are now available under `sgx::enclave::{Builder, Enclave}`. Since `sgx::ioctls` module was used by the `Builder` struct exclusively, it has now been made part of the `enclave` module in `sgx` crate and sealed from the public-facing interface.

@npmccallum @lkatalin I'd really appreciate your input here especially to do with docs, and whether you think moving `{builder.rs, enclave.rs, enclave.S}` from `enarx-keep-sgx` into `sgx` is what we are after.

@lkatalin I've noticed you've touched this code in #761, but just FYI I'll be happy to do all rebases after yours is merged first.

Closes #754 

EDIT: Oh, and I forgot to mention that my suggestion is to have `Builder::build` as the finalizer method rather than `Builder::done`. While from the enclave building perspective, `done` sounds OK (close to `finalize` or something), as Rust dev, I'd expect `build` to consume `self` producing a new struct. Anyhow, I'm not that strong on this though, so if you feel this change is unnecessary/unwanted, I'll happily revert back to `Builder::done`.

### CODE REVIEW

**Comment by kubkon** [comment]
> @kubkon This has conflicts and needs to be rebased. Otherwise, it looks pretty good.

No probs, I'll fix the conflicts and rebase.

**Comment by kubkon** [comment]
> @kubkon This has conflicts and needs to be rebased. Otherwise, it looks pretty good.

@npmccallum done!

File: sgx/src/enclave/mod.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intel SGX Documentation is available at the following link.
+//! Section references in further documentation refer to this document.
+//! https://www.intel.com/content/dam/www/public/emea/xe/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.pdf
+
+#![cfg(feature = "std")]
```
Comment (haraldh) [inline sgx/src/enclave/mod.rs:7]: Isn't this already defined in `sgx/src/lib.rs`

**Comment by npmccallum** [comment]
@kubkon Merging this became a priority, so I did the rebase myself and manually merged it: 7e0f42f9a4f1dbef9369d88cadf2304d0d5b3021

**Comment by kubkon** [comment]
@npmccallum thanks!

**Pull request closed without merging**